### DISCUSSION TITLE
Questions and sharings regarding Debugging

### DISCUSSION THREAD

**Discussion by ramiroaraujo** [johnlindquist/kit#1351] [Q&A]
Since the latests pre-release version the app and especially the debugger works pretty well.
Now, I'm not a fan at all of the devtools for debugging, and I'd rather use the IDE (Webstorm, but VS Code would be great too), so I poked around and I managed to get it working pretty well. Not great, but I think is as good as it's gonna be.

I can setup the debugger via "Attach to Node.js/Chome" to port 51515, and it connects if the Script in debug mode is running. If it's not, I can run it and will connect (I also checked to reconnect automatically).

The bad:
* I need the `debugger` keyword. Regular breakpoints don't work. So no breakpoint conditions for example
* It opens a read-only copy of the file, so any changes I need to perform on the original and re-launch
* It works reliably, but I did happen one time that couldn't connect anymore and gave me some cryptic msg that I can't remember. I had to restart Kit and just in case the IDE.

The good:
* I can debug in the IDE!! My shortcuts, my tools, etc.
* I can inspect and modify runtime variables, work with the current state, evaluate custom multiline expressions, etc.
* Since Kit is now running Typescript files without transpiling, the readonly copy it opens is mostly the same as the original file, and seems to inline on top the imported helper libraries I have around

The only question I have is:
Is this as good as it gets? Or is there a chance to set breakpoints on the IDE, conditional breakpoints, etc?

**Comment by johnlindquist** [comment]
@ramiroaraujo I don't have an answer right now. My best guess is that it's something I can add to the Script Kit VS Code plugin. You can already use the plugin to launch a script, but I _think_ it should be possible that I can trigger the debugger from VS Code as well. I'll dig into it.

**Reply by johnlindquist** [reply]
Oh, and here's a link to the plugin:

https://marketplace.visualstudio.com/items?itemName=johnlindquist.kit-extension

**Reply by ramiroaraujo** [reply]
Awesome! I'd love that option :)
